var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -289,36 +318,66 @@ def write_library_dirs(f, dirs):
                         )
                     )
 
+    def setup_dependent_build_environment(self, env, depspec):
+        self.setup_build_environment(env)
+
     def setup_build_environment(self, env):
         # Tell numpy which BLAS/LAPACK libraries we want to use.
         # https://github.com/numpy/numpy/pull/13132
         # https://numpy.org/devdocs/user/building.html#accelerated-blas-lapack-libraries
         spec = self.spec
 
         # https://numpy.org/devdocs/user/building.html#blas
-        if 'blas' not in spec:
-            blas = ''
-        elif spec['blas'].name == 'intel-mkl' or \
-                spec['blas'].name == 'intel-parallel-studio':
-            blas = 'mkl'
-        elif spec['blas'].name == 'blis':
-            blas = 'blis'
-        elif spec['blas'].name == 'openblas':
-            blas = 'openblas'
-        elif spec['blas'].name == 'atlas':
-            blas = 'atlas'
-        elif spec['blas'].name == 'veclibfort':
-            blas = 'accelerate'
-        else:
-            blas = 'blas'
+        # Also from https://github.com/numpy/numpy/blob/main/numpy/distutils/system_info.py:
+        # The order of finding the locations of resources is the following:
+        #  1. environment variable
+        #  2. section in site.cfg
+        #  3. DEFAULT section in site.cfg
+        #  4. System default search paths (see ``default_*`` variables below).
+        # Only the first complete match is returned.
+        #
+        # The way the detection works, it performs the above scan FOR
+        # EACH different type of lapack. That means that if ATLAS is
+        # installed in a default location, py-numpy will always prefer
+        # it since atlas appears higher in distutils search
+        # order. Adding these environment variables short circuits
+        # this search and gives you the correct behavior.
```

Comment:
This behavior surprises me, I thought that was the whole point of `NPY_BLAS_ORDER`?

Reply:
NPY_BLAS_ORDER wasn't added until 1.17, adding a comment.

Reply:
We also add a patch to add `NPY_BLAS_ORDER` as far back as 1.15. We could just deprecate older versions if you're concerned about supporting them. If this change isn't needed for new versions, I would rather leave it out.

Reply:
I need 1.11 at least for things we're doing here, I can't get a huge code team to budge from it.  Should I protect the env stuff based on the version?

AFIAK there's no harm in setting the environment vars, if something goes wonky setting them ensures that you don't accidentally pick up a system blas when that's not what you intended.

Reply:
Yeah, it's fine to keep them then. If they ever upgrade to modern numpy (I didn't think there would even be issues with backwards compatibility) then we can deprecate those older versions.